app/libs/school_year/archiver.rb

```diff
@@ -0,0 +1,24 @@
+module SchoolYear
+  class Archiver
+    def perform
+      archive_students &&
+      archive_class_rooms
```

Comment:
trouvé ca bizarre le &&
```suggestion
      ActiveRecord::Base.transaction do
        archive_students
        archive_class_rooms
      end
```